spec/que/locker_spec.rb

```diff
@@ -185,6 +185,32 @@ def assert_que_locker_insertion(
     locker.stop!
   end
 
+  it "should create and delete pid file, if passed" do
+    pidfile = 'testfile'
+    locker_settings[:pidfile] = pidfile
+    mock_file = Minitest::Mock.new
+    mock_file.expect :write, nil, [String]
+    mock_open = Minitest::Mock.new
+    mock_open.expect :call, mock_file, [pidfile, ::File::CREAT | ::File::EXCL | ::File::WRONLY]
+    mock_delete = Minitest::Mock.new
+    mock_delete.expect :call, nil, [pidfile]
+
+    File.stub :open, mock_open do
+      File.stub :write, mock_file do
+        locker
+        locker.stop!
+      end
+    end
+
+    File.stub :delete, mock_delete do
+      system("ruby -e 'require \"minitest/autorun\"; exit Minitest.run(ARGV)'")
+    end
```

Comment:
Testing `at_exit` doesn't seem straightforward.

I think I see what you're trying to do here - have a process run the `at_exit` block to delete the file. But doesn't this `system` call run all the tests again? If that even works, I don't think this is a good way to trigger the file deletion - it's indirect and very slow.

How about extracting `#write_pid` into a new class, and testing that in isolation in a similar way to what you were trying to do by using a subprocess, e.g.:

```ruby
# Start a bakground process in a thread:
thread = Thread.new do
  `ruby -r ./lib/que/pid_file.rb -e 'Que::PidFile.create('testfile'); sleep'`
end
# Then check pidfile is present (will require polling and a timeout, since it's beign created asynchronously)
# Then kill the thread, which should close the subprocess
# Then check pidfile is not present
```

Reply:
@ZimbiX No, that doesn't work. When you kill the thread, ruby does not call at_exit. But I changed the thread and now everything works fine.

@oeoeaio Now all tests are green.

Reply:
I'm sure there would have been a way to get something like that to work - like manually sending a particular kill signal to the subprocess's pid. But I think your alternative is fine